domains/primitives/runtime/src/lib.rs

```diff
@@ -195,20 +216,50 @@ sp_api::decl_runtime_apis! {
         /// Returns true if the extrinsic is an inherent extrinsic.
         fn is_inherent_extrinsic(extrinsic: &<Block as BlockT>::Extrinsic) -> bool;
 
-        /// Checks the validity of extrinsic in a bundle.
+        /// Checks the validity of extrinsic.
+        #[changed_in(2)]
+        fn check_transaction_validity(
+            uxt: &<Block as BlockT>::Extrinsic,
+            block_number: NumberFor<Block>,
+            block_hash: <Block as BlockT>::Hash
+        ) -> Result<(), CheckTxValidityError>;
+
+        /// Checks the validity of extrinsic
+        /// v2 can check validity of unsigned and self contained extrinsics as well.
         fn check_transaction_validity(
             uxt: &<Block as BlockT>::Extrinsic,
             block_number: NumberFor<Block>,
             block_hash: <Block as BlockT>::Hash
         ) -> Result<(), CheckTxValidityError>;
 
+        /// Check transaction validity and also perform pre dispatch on transaction.
+        /// This is useful to maintain side effect in the RuntimeApi instance's storage buffer.
+        fn check_transaction_validity_and_do_pre_dispatch(
```

Comment:
We should add also add `#[api_version(2)]` for new api

Reply:
@NingLin-P IIRC as per documentation if `#[api_version(2)]` attribute is for the runtime declaration trait. All new methods comes under version 2. 

https://paritytech.github.io/polkadot-sdk/master/sp_api/macro.decl_runtime_apis.html#runtime-api-trait-versioning